ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsPluginUiTest.java

```diff
@@ -0,0 +1,611 @@
+                .hasTitleText("Static Analysis: 4 warnings")
+                .hasAggregation("FindBugs, CPD, CheckStyle, PMD")
+                .hasNewSize(0)
+                .hasFixedSize(0)
+                .hasReferenceBuild(0);
+
+        reconfigureJobWithResource(job, "build_status_test/build_02");
+
+        Build build = buildJob(job);
+
+        build.open();
+
+        AnalysisSummary analysisSummary = new AnalysisSummary(build, ANALYSIS_ID);
+        assertThat(analysisSummary).isDisplayed()
+                .hasTitleText("Static Analysis: 25 warnings")
+                .hasAggregation("FindBugs, CPD, CheckStyle, PMD")
+                .hasNewSize(23)
+                .hasFixedSize(2)
+                .hasReferenceBuild(1);
+
+        AnalysisResult result = analysisSummary.openOverallResult();
+        assertThat(result).hasActiveTab(Tab.TOOLS).hasTotal(25)
+                .hasOnlyAvailableTabs(Tab.TOOLS, Tab.PACKAGES, Tab.FILES, Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);
+    }
+
+    private IssuesRecorder addRecorder(final FreeStyleJob job) {
+        return job.addPublisher(IssuesRecorder.class, recorder -> {
+            recorder.setTool("CheckStyle");
+            recorder.addTool("FindBugs");
+            recorder.addTool("PMD");
+            recorder.addTool("CPD",
+                    cpd -> cpd.setHighThreshold(8).setNormalThreshold(3));
+            recorder.setEnabledForFailure(true);
+        });
+    }
+
+    /**
+     * Test to check that the issue filter can be configured and is applied.
+     */
+    @Test
+    public void should_filter_issues_by_include_and_exclude_filters() {
+        FreeStyleJob job = createFreeStyleJob("issue_filter/checkstyle-result.xml");
+        job.addPublisher(IssuesRecorder.class, recorder -> {
+            recorder.setTool("CheckStyle");
+            recorder.setEnabledForFailure(true);
+            recorder.addIssueFilter("Exclude categories", "Checks");
+            recorder.addIssueFilter("Include types", "JavadocMethodCheck");
+        });
+
+        job.save();
+
+        Build build = buildJob(job);
+
+        assertThat(build.getConsole()).contains(
+                "Applying 2 filters on the set of 4 issues (3 issues have been removed, 1 issues will be published)");
+
+        AnalysisResult resultPage = new AnalysisResult(build, "checkstyle");
+        resultPage.open();
+
+        IssuesTable issuesTable = resultPage.openIssuesTable();
+        assertThat(issuesTable).hasSize(1);
+    }
+
+    private void reconfigureJobWithResource(final FreeStyleJob job, final String fileName) {
+        job.configure(() -> job.copyResource(WARNINGS_PLUGIN_PREFIX + fileName));
+    }
+
+    /**
+     * Creates and builds a maven job and verifies that all warnings are shown in the summary and details views.
+     */
+    @Test
+    @WithPlugins({"maven-plugin", "analysis-model-api@7.0.4"})
+    public void should_show_maven_warnings_in_maven_project() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_show_maven_warnings_in_maven_project' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)